### PR TITLE
Master wowl get file ges

### PR DESCRIPTION
**Pull request by SimonGenin**
WIP

### CODE REVIEW

**Comment by SimonGenin** [comment]
J'ai mis ça en fourbe dans service mais ça va changer

File: addons/wowl/static/src/libs/download.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    payload = payload[1];
+  }
+
+  if (url && url.length < 2048) {
+    // if no filename and no mime, assume a url was passed as the only argument
+    fileName = url.split("/").pop().split("?")[0];
+    anchor.href = url; // assign href prop to temp anchor
+    if (anchor.href.indexOf(url) !== -1) {
+      // if the browser determines that it's a potentially valid url path:
+      let ajax = new XMLHttpRequest();
```
Comment (kebeclibre) [inline addons/wowl/static/src/libs/download.js:71]: isn't this the fetch api ?

**Review by kebeclibre — COMMENTED** [review]
I suppose this is good enough as the code has been experienced-proofed over the years.
Would not it be possible though, if time allows (and maybe for that piece of code it doesn't....) to see what can be done with fetch ? Or what happens if we factor this to allow it to be low-level testable ?

**Comment by SimonGenin** [comment]
@kebeclibre I actually went for the fetch api at first. But I would have to rewrite everything so every proofed test code goes to the trap, and this is a case where there are apparently many edge cases.

File: addons/wowl/static/src/libs/download.js (hunk 1 above)
Comment (SimonGenin) [reply addons/wowl/static/src/libs/download.js:71]: @kebeclibre Careful, this file is an imported library. I just change some syntax stuff, but I wouldn't change its logic.

File: addons/wowl/static/src/core/download.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  success?: () => void;
+  error?: (error: {
+    message: Stringifiable|String
+    data: {
+      name?: Stringifiable|String,
+      title: Stringifiable|String
+    };
+  }) => void;
+  complete?: () => void;
+}
```
Comment (SimonGenin) [inline addons/wowl/static/src/core/download.ts]: @aab-odoo I have my doubts on this, I don't really know if I should type the callbacks or make them something more premissive.

File: addons/wowl/static/src/core/download.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
+import download from "../libs/download";
+import {_lt, Stringifiable} from "./localization";
```
Comment (SimonGenin) [inline addons/wowl/static/src/core/download.ts]: @aab-odoo I used _lt instead or _t, but I'm not sure it's the way to go.

File: addons/wowl/static/src/types.ts [hunk 4]
```diff
@@ -65,6 +65,7 @@ export interface SessionInfo {
 
 export interface Odoo {
   session_info: SessionInfo;
+  csrf_token?: string;
```
Comment (SimonGenin) [inline addons/wowl/static/src/types.ts:68]: @aab-odoo Optional because it is not present in the mocking. Maybe it should be required ?

File: addons/wowl/static/src/core/download.ts (hunk 3 above)
Comment (FrancoisGe) [reply addons/wowl/static/src/core/download.ts]: I think you need to use "env._t"
